src/main/java/org/folio/mr/service/impl/EcsRequestServiceImpl.java

```diff
@@ -1,50 +1,103 @@
 package org.folio.mr.service.impl;
 
+import static java.lang.String.format;
 import static org.folio.mr.support.ConversionUtils.asString;
 
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
 import org.folio.mr.client.EcsTlrClient;
 import org.folio.mr.domain.dto.EcsTlr;
+import org.folio.mr.domain.dto.User;
+import org.folio.mr.domain.dto.UserPersonal;
+import org.folio.mr.domain.entity.FakePatronLink;
 import org.folio.mr.domain.entity.MediatedRequestEntity;
+import org.folio.mr.repository.FakePatronLinkRepository;
 import org.folio.mr.service.ConsortiumService;
 import org.folio.mr.service.EcsRequestService;
+import org.folio.mr.service.UserService;
 import org.folio.spring.service.SystemUserScopedExecutionService;
 import org.springframework.stereotype.Service;
 
-import lombok.RequiredArgsConstructor;
-import lombok.extern.log4j.Log4j2;
+import java.util.UUID;
 
 @Service
 @Log4j2
 @RequiredArgsConstructor
 public class EcsRequestServiceImpl implements EcsRequestService {
 
+  private final FakePatronLinkRepository fakePatronLinkRepository;
   private final EcsTlrClient ecsTlrClient;
+  private final UserService userService;
+  private final UserCloningServiceImpl userCloningService;
   private final SystemUserScopedExecutionService executionService;
   private final ConsortiumService consortiumService;
 
   @Override
   public EcsTlr create(MediatedRequestEntity mediatedRequest) {
-    log.info("confirm:: creating ECS TLR for mediated request {}", mediatedRequest::getId);
-    return executionService.executeSystemUserScoped(consortiumService.getCentralTenantId(),
-      () -> ecsTlrClient.post(buildEcsTlr(consortiumService.getCurrentTenantId(),
-        mediatedRequest)));
+    log.info("create:: Creating fake patron for mediated request {}", mediatedRequest.getId());
+    String fakeUserId = createFakePatron(mediatedRequest);
+
+    log.info("create:: Creating fake patron link between real patron {} and fake patron {}",
+      mediatedRequest.getRequesterId(), fakeUserId);
+    createFakePatronLink(mediatedRequest.getRequesterId(), fakeUserId);
+
+    log.info("create:: Creating ECS request for fake patron {}, mediated request {}", fakeUserId,
+      mediatedRequest.getId());
+    return createEcsTlr(mediatedRequest, fakeUserId);
+  }
+
+  private String createFakePatron(MediatedRequestEntity mediatedRequest) {
+    User realPatron = userService.fetchUser(asString(mediatedRequest.getRequesterId()));
+    User fakePatron = buildFakePatron(realPatron.getPatronGroup());
+
+    log.info("createFakePatron:: Creating local fake patron");
+    User localFake = userService.create(fakePatron);
+
+    log.info("createFakePatron:: Creating remote fake shadow patron: ID {}", localFake.getId());
+    executionService.executeSystemUserScoped(consortiumService.getCentralTenantId(),
+      () -> userCloningService.cloneUser(localFake, consortiumService.getCurrentTenantId()));
```

Comment:
No need to clone the requester, mod-tlr will do it for you

Reply:
Requester is not cloned